applications/substrate-identity-directory.md

```diff
@@ -103,9 +97,23 @@ We have experience in several private and open source projects aswell as project
 | 2. | Make the web service work offline | Web service can be used in offline mode; the user can specify a local node to which will the service connect. |
  
 
+### Milestone 3 Index data and query
+* **Estimated Duration:** 3 weeks
+* **FTE:**  2
+* **Costs:** 8166 DAI
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 1. | Index data from nodes | Setup [substrate-archive](https://github.com/paritytech/substrate-archive) with postgres for Polkadot and Kusama chains to store required data |
+| 2. | Backend development | Provide endpoints for data queries related to governance and treasury activities |
+| 3. | Fetch data on frontend | Consume provided endpoints and display data on frontend |
+
+Mockup:  
+![Identity activities](https://i.imgur.com/h9wQwXY.png)
+
 ## Future Plans
 
-After finishing milestones 1 and 2 users can utilize the basic application which supports querying by address, index or identity fields. Application has a list page and a single page view with a basic info column fully functional. Users will be able to set their own node endpoint.
+After finishing milestones 1 and 2 users can utilize the basic application which supports querying by address. Application has a list page and a single page view with a basic info column fully functional. Users will be able to set their own node endpoint.
```

Comment:
Why did you remove this? It seems like a very useful feature to me.

Reply:
Accidentally. I have reverted that now.